CONTRIBUTING.md

```diff
@@ -1,20 +1,144 @@
-Thanks for wanting to contribute to Mozilla's Add-ons Linter! You rock! 😊
+Hi! Thanks for wanting to contribute to Mozilla's Add-ons Linter! You rock! 😊
 
-## Submitting a Pull Request
+The linter is used in various stages of [WebExtension](https://developer.mozilla.org/en-US/Add-ons/WebExtensions)
+development and publishing so by contributing you can make that process more awesome.
 
-If you're submitting a pull request, please reference the issue it closes
-inside your pull request. Keep your commits atomic; you should have one
-commit per issue solved.
+Here are links to all the sections in this document:
 
-Add tests for your code and make sure all existing tests pass. If the tests
-fail or you don't maintain 100% test coverage we won't be able to accept your
-pull request.
+<!-- If you change any of the headings in this document, remember to update the table of contents. -->
+<!-- To update the TOC, run the command `npm run gen-contributing-toc` from your root directory and you will auto generate a new TOC. -->
 
-### Tests
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
 
-Our tests include `eslint` style checks; these keep our code consistent.
-You can review the rules in the [`.eslintrc`][eslint] file.
 
-Please run the tests locally with `npm test` before you commit.
+- [Picking an issue](#picking-an-issue)
+- [Installation](#installation)
+  - [Develop all the things](#develop-all-the-things)
```

Comment:
I know that "All the things" is a cute meme but not everyone gets memes, especially non-Western, English-speaking folks. This could be better written as "Development environment".

But actually looking at that section I think it can be removed and folded into testing.